contracts/Market.sol

```diff
@@ -417,7 +417,6 @@ contract Market is PausableAccessPreset {
     for (uint256 i = 0; i < numberOfActiveRemovalsInMarket; i++) {
       uint256 removalId = _activeSupply[_currentSupplierAddress]
         .getNextRemovalForSale();
-      // todo retrieve balances in a single batch call
```

Comment:
I don't believe we can do this in a batch call because that batch call would have to get the balances for the entire supply of the market, which is super wasteful.  seems like we should keep doing it this way and only get the balances for removals as-needed because we might only need to move through 1 or a few of them to fulfill the order

Reply:
think you're right-- unfortunate